chrome/content/zotero/itemTree.jsx

```diff
@@ -1389,24 +1378,12 @@ var ItemTree = class ItemTree extends LibraryTree {
 			var sortStringB = itemB[prop];
 			if (fieldA === undefined) {
 				let firstCreator = Zotero.Items.getSortTitle(sortStringA);
-				if (sortCreatorAsString) {
-					var fieldA = firstCreator;
-				}
-				else {
-					var matches = andEtAlRegExp.exec(firstCreator);
-					fieldA = matches ? matches[0] : '';
-				}
+				fieldA = firstCreator;
 				creatorSortCache[aItemID] = fieldA;
 			}
 			if (fieldB === undefined) {
 				let firstCreator = Zotero.Items.getSortTitle(sortStringB);
-				if (sortCreatorAsString) {
-					var fieldB = firstCreator;
-				}
-				else {
-					matches = andEtAlRegExp.exec(firstCreator);
-					fieldB = matches ? matches[0] : '';
-				}
+				fieldB = firstCreator;
```

Comment:
Does this change the behavior? (I thought I knew what this did, but I can't actually trigger a difference here.)

Reply:
The code that was there tries to cut off the creator’s name at the first `and` or `et al.`, but that isn’t useful - the input is a single creator’s name from the database. So I’m not sure what the idea was originally, but we don’t need it.

Reply:
`firstCreator` here isn't the `firstCreator` string as generated by `_firstCreatorSQL()` — e.g., "Smith et al."?

Reply:
If `sortCreatorAsString` is true, it's `item.firstCreator`, which is the output from `_firstCreatorSQL()`. But that was the code path that would lead to the string *not* being trimmed, I guess because `sortCreatorAsString` means that you want to sort by the entire string, `et al.`s and all.

https://github.com/zotero/zotero/blob/40ae2e32462e55fd1048888c267e7ecd76838c5d/chrome/content/zotero/itemTree.jsx#L1392-L1394

If `sortCreatorAsString` is false, it's `item.sortCreator`, which is just a space-separated list of creator names with no localized delimiters. And on that path, we we *were* trying to trim the `and`s and `et al.`s.

Reply:
So maybe that was a bug and the trimming was supposed to happen if `sortCreatorAsString` is true? But then I wouldn't exactly see the point of the pref. My understanding is that it makes it sort alphabetically based on exactly what you see in the Creator item tree column.